File: crates/livekit-webrtc/libwebrtc-sys/src/rtp_receiver.rs
#[cxx::bridge(namespace = "livekit")]
pub mod ffi {
    unsafe extern "C++" {
        include!("livekit/rtp_receiver.h");
        include!("livekit/media_stream.h");

        type MediaStreamTrack = crate::media_stream::ffi::MediaStreamTrack;
        type RtpReceiver;

        fn track(self: &RtpReceiver) -> UniquePtr<MediaStreamTrack>;

        fn _unique_rtp_receiver() -> UniquePtr<RtpReceiver>; // Ignore
    }
}

unsafe impl Sync for ffi::RtpReceiver {}

unsafe impl Send for ffi::RtpReceiver {}

File: crates/livekit-webrtc/libwebrtc-sys/src/rtp_receiver.cpp
//
// Created by Théo Monnom on 01/09/2022.
//

#include "livekit/rtp_receiver.h"

namespace livekit {
RtpReceiver::RtpReceiver(
    rtc::scoped_refptr<webrtc::RtpReceiverInterface> receiver)
    : receiver_(std::move(receiver)) {}

std::unique_ptr<MediaStreamTrack> RtpReceiver::track() const {
  return MediaStreamTrack::from(receiver_->track());
}

}  // namespace livekit